module/user-activity/src/commonMain/kotlin/org/cru/godtools/shared/user/activity/model/UserActivity.kt

```diff
@@ -31,7 +31,7 @@ data class UserActivity private constructor(
         linksShared = counters[LINK_SHARED] ?: 0,
         languagesUsed = counters.count(LANGUAGE_USED_PREFIX),
         sessions = counters[SESSION] ?: 0,
-        badges = generateBadges(counters),
+        badges = generateBadges(counters).sortedByDescending { it.isEarned },
```

Comment:
evidently boolean sorting is `false` then `true`. so, using a descending sort reverses the order so earned badges are first

Reply:
Do we know if that is a rule or if it is JVM dependent?

Reply:
wrote a test for the sort order, so it should be consistent on android and ios. otherwise I will have to fix the sort logic so the broken test passes ;)

Reply:
this logic is actually not slated to be used on web at all